### PR TITLE
Docs refactor of ingestion. Carries #11541

### PR DESCRIPTION
**Pull request by techdocsmith**
Carries Carries #11541.
Refactor ingestion into smaller topics. Clean up some style here and there.

### Description

This PR breaks down the ingestion landing page topic into smaller topics. Some benefits include:
- Ingestion concepts like the Data Model, Rollup, and Partitioning now have smaller more focused topics.
- The Ingestion spec reference stands alone as a reference topic.
- Should make it easier to add conceptual or example material for a specific concept.

There may be some trade off in topic maintenance. For example if there's an enhancement to rollup, we may need to edit `rollup.md` and `ingestion-spec.md`.

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by techdocsmith** [comment]
Looks like Travis CI failures are not docs related?

**Review by sthetland — APPROVED** [review]
LGTM

File: docs/ingestion/partitioning.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+You can use segment partitioning and sorting within your Druid datasources to reduce the size of your data and increase performance.
+
+One way to partition is to your load data into separate datasources. This is a perfectly viable approach that works very well when the number of datasources does not lead to excessive per-datasource overheads.
```
Comment (jihoonson) [inline docs/ingestion/partitioning.md:29]:
```suggestion
One way to partition is to load data into separate datasources. This is a perfectly viable approach that works very well when the number of datasources does not lead to excessive per-datasource overheads. 
```

**Review by jihoonson — COMMENTED** [review]
Nice work @techdocsmith! LGTM but please fix the error I left a comment about.

**Pull request merged**